src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -0,0 +1,676 @@
+	 * the value contains strings that represent booleans ("true" and "false"),
+	 * numbers ("1" and "1.2") or "null", the function also transform them to
+	 * regular booleans, numbers and `null`.
+	 *
+	 * Example:
+	 *
+	 *     extract_directive_value( 'actions.foo', 'myPlugin' )                      => array( 'myPlugin', 'actions.foo' )
+	 *     extract_directive_value( 'otherPlugin::actions.foo', 'myPlugin' )         => array( 'otherPlugin', 'actions.foo' )
+	 *     extract_directive_value( '{ "isOpen": false }', 'myPlugin' )              => array( 'myPlugin', array( 'isOpen' => false ) )
+	 *     extract_directive_value( 'otherPlugin::{ "isOpen": false }', 'myPlugin' ) => array( 'otherPlugin', array( 'isOpen' => false ) )
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string|true $directive_value   The directive attribute value. It can be `true` when it's a boolean
+	 *                                       attribute.
+	 * @param string|null $default_namespace Optional. The default namespace if none is explicitly defined.
+	 * @return array An array containing the namespace in the first item and the JSON, the reference path, or null on the
+	 *               second item.
+	 */
+	private function extract_directive_value( $directive_value, $default_namespace = null ): array {
+		if ( empty( $directive_value ) || is_bool( $directive_value ) ) {
+			return array( $default_namespace, null );
+		}
+
+		// Replaces the value and namespace if there is a namespace in the value.
+		if ( 1 === preg_match( '/^([\w\-_\/]+)::./', $directive_value ) ) {
+			list($default_namespace, $directive_value) = explode( '::', $directive_value, 2 );
+		}
+
+		/*
+		 * Tries to decode the value as a JSON object. If it fails and the value
+		 * isn't `null`, it returns the value as it is. Otherwise, it returns the
+		 * decoded JSON or null for the string `null`.
+		 */
+		$decoded_json = json_decode( $directive_value, true );
+		if ( null !== $decoded_json || 'null' === $directive_value ) {
+			$directive_value = $decoded_json;
+		}
+
+		return array( $default_namespace, $directive_value );
+	}
+
+
+	/**
+	 * Processes the `data-wp-interactive` directive.
+	 *
+	 * It adds the default store namespace defined in the directive value to the
+	 * stack so it's available for the nested interactivity elements.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_Interactivity_API_Directives_Processor $p               The directives processor instance.
+	 * @param array                                     $context_stack   The reference to the context stack.
+	 * @param array                                     $namespace_stack The reference to the store namespace stack.
+	 */
+	private function data_wp_interactive_processor( WP_Interactivity_API_Directives_Processor $p, array &$context_stack, array &$namespace_stack ) {
+		// In closing tags, it removes the last namespace from the stack.
+		if ( $p->is_tag_closer() ) {
+			return array_pop( $namespace_stack );
```

Comment:
This method apparently shouldn't return anything.

```suggestion
			array_pop( $namespace_stack )
			return;
```

Reply:
Fixed in https://github.com/WordPress/wordpress-develop/pull/5953/commits/97d4c98f28ba9e90f2e6bdb4844b31f0ae670183.